### ISSUE TITLE
Pre-phased haplotypes lead to inconsistent contig results

### ISSUE THREAD

**Issue by pengelgau** [marbl/canu#1400]
Some background on my goal: 
I'm trying to haplotype a ~6kb locus that contains my gene of interest from a variety of apple cultivars. I used PCR to amplify my region, and multiplexed the run. I've since separated my reads by genotype via Porechop and then have also separated my reads by haplotype using WhatsHap using known SNPs from Sanger and past cloning work. What I'd like to do next is to assemble these reads into a "draft haplotype" with canu and then polish this up with Nanopolish. 

As I run canu I often get about one-third of my 48 haplotypes (all run separately) not produce a contig. I've experimented and used a random unassembled read for downstream polishing but there are noticeable errors that are not present when I use a proper contig. 

I'm not sure why some are generating them and some aren't because the inputs going in are all very similar with at least 200X coverage, often times much, much more. I wonder if there is some modification or setting that I'm missing that will correct this. I've experimented with different options but they never produce a contig. Below is my latest trial, experimenting with changing the overlap and read lengths (I've been using 1.8 for everything, running on MacOS, locally):

``` canu -p JGO1alldraft2 -d JGO1alldraft2 genomeSize=5600 -nanopore-raw JGO/JGO1all.fastq contigFilter="2 0 1.0 0.5 0" correctedErrorRate=0.12 minOverlapLength=1000 minReadLength=4000```

Attached are reports from the two haplotypes of my JGO genotype. One of which produced a contig (JGO2), the other didn't (JGO1).
[JGO1alldraft2.txt](https://github.com/marbl/canu/files/3321220/JGO1alldraft2.txt)
[JGO2alldraft1.txt](https://github.com/marbl/canu/files/3321222/JGO2alldraft1.txt)

Thanks for taking the time to look at my issue. I really appreciate it.

**Comment by skoren** [comment]
Similar to other amplicon-related issues, #1313, #1322, #1379, there really isn't anything to assemble. You could probably just use a single corrected read that's 5.6 kb (which you have a lot of in both reports) and it would be a good representative of the amplicon.

You could also run with adjusted ```contigFilter="2 0 1.0 0.5 0"``` and/or downsampling the coverage (```readSamplingCoverage=100```).

**Comment by skoren** [comment]
Any update on this?

**Comment by skoren** [comment]
Idle

**Comment by pengelgau** [comment]
Totally forgot to reply on this, sorry about that.

When finding an assembled read for downstream application is turns out that not all of those reads are equal. I at first was always taking the first read in the file for consistency. But when looking at and aligning all of them I saw that for whatever reason that first read was significantly different from the others and was a poor representative. Major differences being usually large structural indels that were different from the rest of that haplotype. This occurred about 20% of the time. 

What I then did was screen through and select the best representative. Something that had the same indels, SNPS, etc as the rest, but was also free of known erroneous indels that I had observed when comparing to Sanger data or the reference genome. However, when using this contig as a 'reference genome' for nanopolish, some of those errors that I had purposely tried to screen out would be put back in by nanopolish. That seems to me to be a nanopolish issue. Anyways I've gotten results that I feel are satisfactory but that still carry an AS IS disclaimer with regards to large sections.
